data/test/language-snippets/format/list.sw

```diff
@@ -0,0 +1,306 @@
+// - HELPERS: consP, getLenPart, setLenPart, to1numA, getLenA
+// - TESTS: assert, testLIST, testLIST_BIG, testLIST_ALL
+//
+/*******************************************************************/
+/*                              TYPE                               */
+/*******************************************************************/
+// Type of list of ints - the inner ints can themselves be lists,
+// so *any* type can be encoded inside.
+//
+// It is the callers responsibility to make sure a program using this
+// "type" is type safe. Notably 2 == [0] != [] == 0 but [] !! x == 0.
+tydef ListI = Int end
+
+
+/*******************************************************************/
+/*                             LAYOUT                              */
+/*******************************************************************/
+// The length of a number is kept in the header and split into 7bit
+// chunks each prefixed by 1bit that marks if the byte is last in
+// the header (0=YES).
+/* EXAMPLE - [short_x,long_y] - concretly e.g. [42, 2^(2^7)]
+
+0   < len short_x < 2^7
+2^7 < len long_y  < 2^14
+
+cons short_x   $          cons long_y          $ nil
+vvvvvvvvvvvv       vvvvvvvvvvvvvvvvvvvvvvvvv     vvv
+  0|len x|x    |   1|len y%2^7|0|len y/2^7|y   |  0
+^^^^^^^^^^^^       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  head                         tail
+*/
+/*******************************************************************/
+/*                         ARITH PRIMITIVES                        */
+/*******************************************************************/
+// bitlength of a number (shifting by one)
+def naive_len: Int -> Int -> Int 
+  = \acc. \i.
+  if (i == 0) {acc} {naive_len (1 + acc) (i / 2)}
+end
+
+
+// modulus function (%)
+def mod: Int -> Int -> Int = \i. \m. i - m * (i / m) end
+
+
+// f X Y = Y * 2^(-X)
+def shiftL: Int -> Int -> Int = \s. \i. i / 2 ^ s end
+
+
+// f X Y = Y * 2^(X)
+def shiftR: Int -> Int -> Int = \s. \i. i * 2 ^ s end
+
+
+// shift by (8bit) bytes
+def shiftLH: Int -> Int -> Int = \s. shiftL (s * 8) end
+
+def shiftRH: Int -> Int -> Int = \s. shiftR (s * 8) end
+
+
+// bitlength of a number using an accumulator (shifting by 64)
+def len_accum: Int -> Int -> Int 
+  = \acc. \i.
+  let next = i / 2 ^ 64 in
+  if (next == 0) {naive_len acc i} {len_accum (acc + 64) next}
+end
+
+
+// bitlength of a number (uses an accumulator and shifts by 64 bits)
+def len: Int -> Int = len_accum 0 end
+
+
+/*******************************************************************/
+/*                       helper functions                          */
+/*******************************************************************/
+def getLenPart: Int -> Int = \i. mod (i / 2) (2 ^ 7) end
+
+def setLenPart: Int -> Int = \i. 2 * mod i (2 ^ 7) end
+
+
+// Split length into 7-bit parts and prefix all but last with 1
+def to1numA: Int -> (Int * Int) 
+  = \i.
+  let nextPart = shiftL 7 i in
+  if (nextPart == 0) {
+
+    // last part
+    (2 * i, 1)
+  } {
```

Comment:
@byorgey, this stands out to me. It seems we are adding an extra line before comments, but that leads to weird cases like this one. Also, the space between definitions is either one or two lines, depending on whether there is a comment before it.

Reply:
Hmm, I don't remember why I originally did it that way, but I agree the extra line before comments does not look good.  Removed!